plotly/plotly/plotly.py

```diff
@@ -1457,6 +1448,76 @@ def _send_to_plotly(figure, **plot_options):
     return r
 
 
+def create_animations(fig, kwargs, payload):
```

Comment:
🚨 💀 😨 I think you should *not* do figure splitting (i.e., exchanging raw data for referenced data). Please take my word on it. It's not worth it. It's *extremely* complicated (Please take my word on it).

^^ Seriously, take my word on it.

Users should be able to do something like:

```
grid = Grid()
grid.add_column('time_data', [1, 2, 3, 4])  # something like this...
grid.add_column('voltage', [120, 119, 122, 121])  # something like this...
# somehow, someway, add data to this grid and make it easy for users to pull out the reference.
# try looking in https://github.com/plotly/plotly.py/blob/master/plotly/grid_objs/grid_objs.py

figure.data[0].x = grid.time_data
figure.frames[0]['data'][0] = grid.time_data  # Not sure if this is right, just for example

create_animations(figure, ...)
```

AFAIK, you don't need to worry *too* much about checking for raw data in the figure. The api will catch you and the request will just fail.

Let's fully shift the responsibility to the user. If we get into the game of guessing, we'll just end up guessing wrong and creating bugs.